web/src/people.js

```diff
@@ -1285,6 +1285,36 @@ export function report_late_add(user_id, email) {
     }
 }
 
+function make_user(user_id, email, full_name) {
+    return {
+        user_id,
+        email,
+        full_name,
+        is_active: false,
+        is_admin: false,
+        is_owner: false,
+        is_guest: false,
+        is_bot: false,
+        is_moderator: false,
+        is_billing_admin: false,
+        // We explicitly don't set `avatar_url` for fake person objects so that fallback code
+        // will ask the server or compute a gravatar URL only once we need the avatar URL,
+        // it's important for performance that we not hash every user's email to get gravatar URLs.
+        avatar_url: undefined,
+        avatar_version: 0,
+        timezone: "",
+        date_joined: "",
```

Comment:
This default value fails this code path: https://github.com/zulip/zulip/blob/8e63c0e42c8b948b64b00bfbf7e711b1c41814d8/web/src/popovers.js#L246-L251
The popover for spectator users assumes that the date joined is valid.

Reply:
Perhaps we can just set it to the current time when the user is late added?

Reply:
I think it's probably better to check `is_missing_server_data` and skip rendering `date_joined` in this case.

Reply:
Yeah I think we can extend this if condition to check for `is_missing_server_data`.